React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+# Chakra UI
+
+Chakra UI is a simple, modular and accessible component library that gives you the building blocks you need to build your React applications.
+
+## Getting started
+
+### Installation:
+
+```
+npm i @chakra-ui/react @emotion/react @emotion/styled framer-motion
+```
+
+### Setup provider:
+
+For Chakra UI to work correctly, you need to set up the `ChakraProvider` at the root of your application.
+
+Example:
+
+```jsx
+import { ChakraProvider } from "@chakra-ui/react";
+
+function App({ Component, pageProps }: AppProps): ReactElement {
+  return (
+    <ChakraProvider theme={theme}>
+      <Component {...pageProps} />
+    </ChakraProvider>
+  );
+}
+
+export default App;
+```
+
+Note:
+
+- For Next.js, you need to set this up in `pages/_app.tsx`
+- For Create React App, you need to set this up in `index.tsx`
+
+### Style props
+
+Style props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components.
+
+Alternative to that is using sx prop that takes an object with style rules.
+
+```tsx
+// shorthand width prop
+<Box w="400px">Lorem Ipsum is simply dummy text</Box>
+
+// sx prop
+<Box sx={{ width: '400px'}}>Lorem Ipsum is simply dummy text</Box>
+```
+
+For more about style props read [docs](https://chakra-ui.com/docs/features/style-props)
+
+#### The `as` prop
```

Comment:
This is called `polymorphic prop` and it's actually a feature of Emotion https://emotion.sh/docs/styled#as-prop borrowed from styled-components https://styled-components.com/docs/api#as-polymorphic-prop.
You should mention it in this section.